brave/ui/app/ducks/metamask/metamask.js

```diff
@@ -1,11 +1,23 @@
+const extend = require('extend')
+const actions = require('../../store/actions')
 const reduceMetamask = require('../../../../../ui/app/ducks/metamask/metamask')
 
 module.exports = function (state, action) {
   const newState = reduceMetamask(state, action)
+
   newState.batTokenAdded = newState.batTokenAdded || false
+  newState.rewardsDisclosureAccepted = newState.rewardsDisclosureAccepted || false
+
+  switch (action.type) {
+    case actions.SET_BAT_TOKEN_ADDED:
+      return extend(newState, {
+        batTokenAdded: action.value,
+      })
 
-  if (action.type === 'SET_BAT_TOKEN_ADDED') {
-    newState.batTokenAdded = action.value
+    case actions.REWARDS_DISCLOSURE_ACCEPTED:
```

Comment:
Is this still needed if the state is handled via background? Seems like we never actually dispatch this and just use `actions. setRewardsDisclosureAccepted` instead, which routes through the preferences controller.

Reply:
nope, not needed